server/routerlicious/packages/services/src/throttlerHelper.ts

```diff
@@ -96,21 +97,27 @@ export class ThrottlerHelper implements IThrottlerHelper {
         const timeSinceLastCooldownInMs = now - throttlingMetric.lastCoolDownAt;
         // replenish "tokens" at most once per minCooldownInterval
         if (timeSinceLastCooldownInMs > this.minCooldownIntervalInMs) {
-            return Math.floor(timeSinceLastCooldownInMs * this.rateInOperationsPerMs);
+            const tokensToReplenish = Math.floor(timeSinceLastCooldownInMs * this.rateInOperationsPerMs);
+            // don't let the bucket overflow
+            if (tokensToReplenish + throttlingMetric.count > this.operationBurstLimit) {
+                const tokensToFillBucket = this.operationBurstLimit - throttlingMetric.count;
+                return tokensToFillBucket;
```

Comment:
return this.operationBurstLimit - throttlingMetric.count;

small optimization.